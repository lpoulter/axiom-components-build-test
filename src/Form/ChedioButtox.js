import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import './ChedioButtox.css';

export default class ChedioButtox extends Component {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string.isRequired,
    disabled: PropTypes.bool,
    indeterminate: PropTypes.bool,
    inputType: PropTypes.oneOf(['checkbox', 'radio']).isRequired,
    invalid: PropTypes.bool,
    onClick: PropTypes.func,
    title: PropTypes.string,
  };

  render() {
    const {
      children,
      className,
      disabled,
      inputType,
      invalid,
      title,
      onClick,
      indeterminate,
      ...rest
    } = this.props;

    const classes = classnames(className, {
      [`${className}--disabled`]: disabled,
      [`${className}--invalid`]: invalid,
      [`${className}--indeterminate`]: indeterminate,
    });

    const handleClick = onClick && (e => {
      if (e.target.tagName !== 'INPUT') {
        e.stopPropagation();
        return;
      }
      onClick(e);
    });

    return (
      <Base
          Component="label"
          className={ classes }
          onClick={ handleClick }
          space="x2"
          title={ title }>
        <input { ...rest }
            className={ `${className}__input` }
            disabled={ disabled }
            type={ inputType } />

        <span className={ `${className}__indicator` } />
        { children && <span className={ `${className}__label` }>{ children }</span> }
      </Base>
    );
  }
}
