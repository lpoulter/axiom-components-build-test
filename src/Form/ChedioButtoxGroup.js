import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Base from '../Base/Base';

export default class ChedioButtoxGroup extends Component {
  static propTypes = {
    children: PropTypes.node,
  };

  render() {
    const { children, ...rest } = this.props;

    return (
      <Base { ...rest } className="ax-chedio-buttox-group">
        { children }
      </Base>
    );
  }
}
