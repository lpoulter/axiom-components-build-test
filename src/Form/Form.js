import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Base from '../Base/Base';
import Validation from '../Validation/Validation';

export default class Form extends Component {
  static propTypes = {
    children: PropTypes.node,
    /**
     * A function called with form event and an error if validation
     * is present.
     *
     * @param {object} event Standard event form object
     * @param {string|null} error Error from Validation if present otherwise null.
     */
    onSubmit: PropTypes.func,
    /** See Validation[requiredError] */
    requiredError: PropTypes.string,
  };

  handleSubmit(event, getUpdatedValidation) {
    const { onSubmit } = this.props;
    const error = getUpdatedValidation();

    if (onSubmit) {
      onSubmit(event, error);
    }
  }

  render() {
    const { children, requiredError, ...rest } = this.props;

    return (
      <Validation requiredError={ requiredError }>
        { (getUpdatedValidation) =>
          <Base { ...rest }
              Component="form"
              className="ax-form"
              onSubmit={ (e) => this.handleSubmit(e, getUpdatedValidation) }>
            { children }
          </Base>
        }
      </Validation>
    );
  }
}
