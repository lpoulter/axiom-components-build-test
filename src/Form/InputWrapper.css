.ax-input,
.ax-textarea {
  width: 100%;
  min-width: 0;
  color: inherit;
  appearance: none;

  &::placeholder {
    opacity: 1;
    color: var(--color-theme-text--disabled);
  }

  &:focus {
    outline: none;

    &::placeholder {
      color: transparent;
    }
  }
}

.ax-input__wrapper {
  display: flex;
  position: relative;
  flex: 1 1 auto;
  flex-direction: column;
}

.ax-input__wrapper--inline {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.ax-input__wrapper--inline .ax-input__icon-container {
  margin-left: var(--spacing-grid-size--x2);
}

.ax-input__hint-wrapper {
  display: flex;
  justify-content: flex-end;
}

.ax-input__hint-wrapper .ax-input__wrapper--label {
  flex-grow: 1;
}

.ax-input__hint {
  flex-shrink: 0;
}

.ax-input__wrapper--inline .ax-input__hint-wrapper {
  align-items: center;
}

.ax-input__wrapper--inline .ax-input__hint {
  margin-left: var(--spacing-grid-size--x1);
}

.ax-input__wrapper--target .ax-input {
  cursor: pointer;
}
