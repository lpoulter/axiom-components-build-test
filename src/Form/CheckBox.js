import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Validate from '../Validation/Validate';
import ChedioButtox from './ChedioButtox';

export default class CheckBox extends Component {
  static propTypes = {
    /** Checked state of the input */
    checked: PropTypes.bool,
    /** Label that is placed next to the input */
    children: PropTypes.node,
    /** Disables interactions and applies styling */
    disabled: PropTypes.bool,
    /** See Validate[error] */
    error: PropTypes.func,
    /** Partially checked state of the input */
    indeterminate: PropTypes.bool,
    /** Applies styling to indicate the users input was invalid */
    invalid: PropTypes.bool,
    /** Name of the group the input belongs to */
    name: PropTypes.string,
    /** See Validate[patterns] */
    patterns: PropTypes.array,
    /** See Validate[required] */
    required: PropTypes.bool,
    /** Text that is displayed on hover */
    title: PropTypes.string,
  };

  render() {
    const {
      checked,
      children,
      disabled,
      error,
      invalid,
      name,
      indeterminate,
      patterns,
      required,
      ...rest
    } = this.props;

    return (
      <Validate
          error={ error }
          patterns={ patterns }
          required={ required }
          value={ checked }>
        { (isValid) =>
          <ChedioButtox
              { ...rest }
              checked={ !!(checked || indeterminate) }
              className="ax-checkbox"
              disabled={ disabled }
              indeterminate={ indeterminate && !checked }
              inputType="checkbox"
              invalid={ invalid || isValid === false }
              name={ name }>
            { children }
          </ChedioButtox>
        }
      </Validate>
    );
  }
}
