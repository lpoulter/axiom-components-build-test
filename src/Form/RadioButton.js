import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Validate from '../Validation/Validate';
import ChedioButtox from './ChedioButtox';

export default class RadioButton extends Component {
  static propTypes = {
    /** Checked state of the input */
    checked: PropTypes.bool,
    /** Label that is placed next to the input */
    children: PropTypes.node,
    /** Disables interactions and applies styling */
    disabled: PropTypes.bool,
    /** See Validate error */
    error: PropTypes.func,
    /** Applies styling to indicate the users input was invalid */
    invalid: PropTypes.bool,
    /** Name of the group the input belongs to */
    name: PropTypes.string,
    /** See Validate[patterns] */
    patterns: PropTypes.array,
    /** See Validate[required] */
    required: PropTypes.bool,
    /** Text that is displayed on hover */
    title: PropTypes.string,
  };

  render() {
    const {
      checked,
      children,
      disabled,
      error,
      invalid,
      name,
      patterns,
      required,
      ...rest
    } = this.props;

    return (
      <Validate
          error={ error }
          patterns={ patterns }
          required={ required }
          value={ checked }>
        { (isValid) =>
          <ChedioButtox
              { ...rest }
              checked={ checked }
              className="ax-radio"
              disabled={ disabled }
              inputType="radio"
              invalid={ invalid || isValid === false }
              name={ name }>
            { children }
          </ChedioButtox>
        }
      </Validate>
    );
  }
}
