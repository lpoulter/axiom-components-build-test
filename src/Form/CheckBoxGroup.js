import PropTypes from 'prop-types';
import React, { Component } from 'react';
import ChedioButtoxGroup from './ChedioButtoxGroup';

export default class CheckBoxGroup extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };
  render() {
    const { children, ...rest } = this.props;

    return (
      <ChedioButtoxGroup { ...rest }>
        { children }
      </ChedioButtoxGroup>
    );
  }
}
