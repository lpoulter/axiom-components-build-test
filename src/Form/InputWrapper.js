import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Base from '../Base/Base';
import UsageHintComponent from '../UsageHint/UsageHint';
import './InputWrapper.css';
import classnames from 'classnames';

const labelSizeMap = {
  small: 'small',
  medium: 'body',
  large: 'large',
};

export default class InputWrapper extends Component {
  static propTypes = {
    children: PropTypes.node,
    disabled: PropTypes.bool,
    hasFocus: PropTypes.bool,
    inlineLabel: PropTypes.bool,
    inputType: PropTypes.string,
    invalid: PropTypes.bool,
    isTarget: PropTypes.bool,
    isValid: PropTypes.bool,
    label: PropTypes.node,
    labelSpace: PropTypes.oneOf(['x0', 'x1', 'x2', 'x3', 'x4', 'x5', 'x6', 'x8']),
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    space: PropTypes.string,
    style: PropTypes.string,
    usageHint: PropTypes.node,
    usageHintPosition: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
    valid: PropTypes.bool,
  };

  static defaultProps = {
    size: 'medium',
    space: 'x4',
    labelSpace: 'x2',
    inlineLabel: false,
  };

  render() {
    const { children, inlineLabel, disabled, hasFocus, invalid, isTarget, isValid, label, labelSpace, size, space, style, usageHint, usageHintPosition, valid, ...rest } = this.props;
    const classes = classnames('ax-input__wrapper', {
      'ax-input__wrapper--target': isTarget,
      'ax-input__wrapper--inline': inlineLabel,
    });

    const innerClasses = classnames('ax-input__icon-container', {
      [`ax-input__icon-container--${size}`]: size,
      [`ax-input__icon-container--${style}`]: style,
      'ax-input__icon-container--disabled': disabled,
      'ax-input__icon-container--focused': hasFocus,
      'ax-input__icon-container--valid': valid,
      'ax-input__icon-container--invalid': invalid || isValid === false,
    });

    const inputWrapperSpace = inlineLabel ? 'x0' : labelSpace;

    return (
      <Base { ...rest }
          Component="label"
          className={ classes }
          space={ space }>
        {
          (label || usageHint) && (
            <Base className="ax-input__hint-wrapper" space={ inputWrapperSpace }>
              {
                label && (
                  <Base className="ax-input__wrapper--label" textSize={ labelSizeMap[size] }>
                    { label }
                  </Base>
                )
              }

              {
                usageHint && (
                  <div className="ax-input__hint">
                    <UsageHintComponent position={ usageHintPosition }>
                      { usageHint }
                    </UsageHintComponent>
                  </div>
                )
              }
            </Base>
          )
        }
        <div className={ innerClasses }>
          { children }
        </div>
      </Base>
    );
  }
}
