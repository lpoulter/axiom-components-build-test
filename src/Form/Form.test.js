import React from 'react';
import renderer from 'react-test-renderer';
import Form from './Form';

const getComponent = (props = {}) =>
  renderer.create(
    <Form { ...props }>Lorem ipsum</Form>
  );

describe('Form', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
