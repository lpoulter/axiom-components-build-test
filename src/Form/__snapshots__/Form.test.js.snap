// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders with defaultProps 1`] = `
<form
  className="ax-form"
  onSubmit={[Function]}
>
  Lorem ipsum
</form>
`;
